### PR TITLE
Support InstantStyle

### PR DESCRIPTION
**Pull request by haofanwang**
This PR supports for our recent work [InstantStyle](https://github.com/InstantStyle/InstantStyle) in native diffusers API. The idea has also been discussed at #7534 .

The modifications are mainly about IP-Adapter loader, allowing users to specify target blocks used for image feature injection. After merged, InstantStyle can be achieved via following way

```
pipe.load_ip_adapter(pretrained_model_name_or_path_or_dict="./", 
                     subfolder="sdxl_models", 
                     weight_name="ip-adapter_sdxl.bin",
                     image_encoder_folder=image_encoder_path,
                     target_blocks=["block"]
                    )
```

### CODE REVIEW

**Comment by yiyixuxu** [comment]
@asomoza can you do a final review and test this out?

File: src/diffusers/loaders/unet.py [hunk 1]
```diff
@@ -864,11 +864,19 @@ def _convert_ip_adapter_attn_to_diffusers(self, state_dicts, low_cpu_mem_usage=F
                         num_image_text_embeds += [state_dict["image_proj"]["latents"].shape[1]]
 
                 with init_context():
+
+                    selected = False
+                    for block_name in target_blocks:
+                        if block_name in name:
+                            selected = True
+                            break
```
[the remark this answers is not in the document]
Comment (ResearcherXman) [reply src/diffusers/loaders/unet.py]: Yes, it is much clearer. Updated.

**Comment by haofanwang** [comment]
I believe we can set different weights for each block just like LoRA dict, in this case, we don't need to reload modules. Will update soon.

**Comment by ivanprado** [comment]
Cool @haofanwang! What would be the mechanism to subtract the CLIP text embedding from the image embedding with diffusers? This is done [here](https://github.com/InstantStyle/InstantStyle/blob/main/ip_adapter/ip_adapter.py#L165) in the original implementation.

**Comment by yiyixuxu** [comment]
we will merge this one soon https://github.com/huggingface/diffusers/pull/7499
cool to rebase here? or is it easier the other way around?

**Comment by haofanwang** [comment]
Will rebase on #7499.

**Comment by okaris** [comment]
Is it possible to load multiple ip adapters at the same time?
ip-adapter + instantid
instantid + instantstyle

I would appreciate an example that shows how we could load/unload different adapters seperately or together.

Thanks!

**Comment by haofanwang** [comment]
@yiyixuxu @asomoza Our teammate has made a new PR. This PR will be closed soon.

**Comment by haofanwang** [comment]
> Is it possible to load multiple ip adapters at the same time? ip-adapter + instantid instantid + instantstyle
> 
> I would appreciate an example that shows how we could load/unload different adapters seperately or together.
> 
> Thanks!

This is another problem. We can support it later.

**Comment by haofanwang** [comment]
> Cool @haofanwang! What would be the mechanism to subtract the CLIP text embedding from the image embedding with diffusers? This is done [here](https://github.com/InstantStyle/InstantStyle/blob/main/ip_adapter/ip_adapter.py#L165) in the original implementation.

This can be natively achieved already. We will show how to do it once this PR merged.

**Pull request closed without merging**